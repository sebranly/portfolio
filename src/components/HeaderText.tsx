import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { AUTHOR_NAME } from '../constants/general';

export interface HeaderTextProps {
  className?: string;
}

const HeaderText: React.FC<HeaderTextProps> = (props) => {
  const { className } = props;
  const { t } = useTranslation();

  return (
    <div className={className}>
      <h1 className="font-bold sm:text-6xl text-2xl">{t('header.welcome')}</h1>
      <h2 className="sm:text-4xl text-2xl">
        {t('header.author.prefix')}
        <div className="italic inline">{AUTHOR_NAME}</div>
        {t('header.author.suffix')}
      </h2>
      <div className="sm:text-2xl text-xl">{t('header.introduction')}</div>
    </div>
  );
};

export { HeaderText };
