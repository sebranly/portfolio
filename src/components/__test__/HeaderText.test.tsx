import * as React from 'react';
import { render } from '@testing-library/react';

import { HeaderText } from '../HeaderText';

test('HeaderText', () => {
  const createComponent = () => <HeaderText />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
