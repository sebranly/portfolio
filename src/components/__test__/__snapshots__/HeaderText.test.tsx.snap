// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderText 1`] = `
NodeList [
  <div>
    <h1
      class="font-bold sm:text-6xl text-2xl"
    >
      header.welcome
    </h1>
    <h2
      class="sm:text-4xl text-2xl"
    >
      header.author.prefix
      <div
        class="italic inline"
      >
        sebranly
      </div>
      header.author.suffix
    </h2>
    <div
      class="sm:text-2xl text-xl"
    >
      header.introduction
    </div>
  </div>,
]
`;
